osu.Game/Tests/Visual/Multiplayer/TestMultiplayerClient.cs

```diff
@@ -495,6 +505,7 @@ public async Task RemoveUserPlaylistItem(int userId, long playlistItemId)
                 throw new InvalidOperationException("Attempted to remove an item which has already been played.");
 
             serverSidePlaylist.Remove(item);
+            serverSideAPIRoom.Playlist.RemoveAll(i => i.ID == item.ID);
```

Comment:
The fact that `serverSidePlaylist` and `serverSideAPIRoom.Playlist` coexist - and need to be synced together like this - makes me nervous that there are more opportunities for a desync somewhere in these operations. In fact it looks to me like for instance, `EditUserPlaylistItem()` may be forgetting to change `serverSideAPIRoom.Playlist`, as it's changing `serverSidePlaylist`:

https://github.com/ppy/osu/blob/f92a31cd3994c59fa6e4a909180f10868f58c0c2/osu.Game/Tests/Visual/Multiplayer/TestMultiplayerClient.cs#L471

I'm not sure what the solution here is because it doesn't look like you can easily remove one in favour of using the other exclusively. At least probably a good idea to read through all of the other operations on `serverSidePlaylist` and make sure consistency is being upheld.

Reply:
Yeah I don't know what the solution to that would be. My only suggestion would be to store the entirety of the `MultiplayerRoom`. Fixed `EditUserPlaylistItem`.